### ISSUE TITLE
[Design] Address Arch Board Feedback from .NET/Python Reviews

### ISSUE THREAD

**Issue by annelo-msft** [Azure/azure-sdk-for-net#12476]
- [ ] follow up if there could be something labeled that is not text when thinking of naming for `FieldText`. Should this be `FieldData` or alternate naming? Otherwise property/model naming difference between languages is okay => https://github.com/Azure/azure-sdk-for-net/issues/13036
- [ ] `RequestedOn` and `CompletedOn` should be renamed to `TrainingStartedOn` and `TrainingCompletedOn` => https://github.com/Azure/azure-sdk-for-net/issues/13034
- [x] possibly add training duration - need to understand if this is something users want/need
- [x] KC doesn't like `submodels` - should we brainstorm naming for the property/model here some more?
- [x] KC didn't like `CustomFormModel` and preferred this to be something like `TrainingResult`
- [x] For receipt samples conversion methods, throw exception instead of returning null => https://github.com/Azure/azure-sdk-for-net/issues/13000
- [ ] Rename parameters `FromUrl` to `FromUri` =>https://github.com/Azure/azure-sdk-for-net/issues/13034
- [ ] Rename parameter `accessToken` on `CopyAuthorization.FromJson` as the string contains more information => was pointed out by @ellismg after the meeting => https://github.com/Azure/azure-sdk-for-net/issues/13034
- [ ] Reconsider options overload: => https://github.com/Azure/azure-sdk-for-net/issues/12701
  - Either remove `FromUrl` from the methods so we have 3 overloads from the same method
  - Remove the overload that takes options and add options as a parameter to stream and url methods
- [ ] need to make sure we do support local file paths for containers  => #12701

**Comment by maririos** [comment]
>possibly add training duration

We would like to understand more if this is something the user will like to consume before implementing it. Because it will be an additive property, it won't be a breaking change, and therefore we decided not to implement it for P4.

>KC doesn't like submodels - should we brainstorm naming for the property/model here some more?

The v-team looked into this and because there isn't a better proposal and UX Study showed that users could navigate the property, we decided to leave it as is

> KC didn't like CustomFormModel and preferred this to be something like TrainingResult

v-team looked into this and decided to leave the name `CustomFormModel` and rename the `training` specific properties. At the same time, the name customForm is used in the `FormRecognizerClient` which helps the user understand which method to use and where the `modelId` comes from.
